src/redis/lua/cjson/lua_cjson.c

```diff
@@ -1226,9 +1226,21 @@ static void json_process_value(lua_State *l, json_parse_t *json,
     case T_STRING:
         lua_pushlstring(l, token->value.string, token->string_len);
         break;;
-    case T_NUMBER:
-        lua_pushnumber(l, token->value.number);
-        break;;
+    case T_NUMBER: {
+        double num = token->value.number;
+        double intpart;
+        /* Check if legacy float mode is enabled via global variable */
+        lua_getglobal(l, "__dfly_legacy_float__");
```

Comment:
I agree that now it's much cleaner. But I must ask @dranikpg  @vyavdoshenko  why not use lua_pushinteger unconditionally if the number is integer ?

Reply:
in other words, why do we need `__dfly_legacy_float__` mode at all?

Reply:
what do you mean? we do check it there: `fractpart = modf(d, &intpart);`

Reply:
```
 if (abs(fractpart) < kConvertEps && intpart < double(std::numeric_limits<lua_Integer>::max()) &&
      intpart > std::numeric_limits<lua_Integer>::min())
    lua_pushinteger(lua_, static_cast<lua_Integer>(d));
  else
    lua_pushnumber(lua_, d);
```